common/web/gesture-recognizer/src/engine/headless/touchpointCoordinator.ts

```diff
@@ -109,55 +119,84 @@ export class TouchpointCoordinator<HoveredItemType, StateToken=any> extends Even
+      const modelingSpinupResults = await modelingSpinupPromise;
+
+      if(modelingSpinupResults.sustainModeWithoutMatch) {
+        /* May need to do a state-token change check & update the item; an `awaitNested` 'complete' action
+         * may have been pending in the meantime that could have triggered a change.
+         *
+         * (The MatcherSelector's state token will not have been updated b/c it will have already been popped,
+         * and because it's popped, it should not be responsible for managing the new GestureSource -
+         * including shifts in state token.)
+         *
+         * Current actual use-case:  deferred modipress due to ongoing flick, auto-completed by new incoming touch.
+         */
+        touchpoint.path.coords.forEach((coord) => {
+          coord.stateToken = this.stateToken;
+          coord.item = touchpoint.currentRecognizerConfig.itemIdentifier(coord, null);
+        });
+        continue;
+      } else {
+        selectionPromise = modelingSpinupResults.selectionPromise;
+        break;
+      }
+
+      // Can only happen if a `sustainWhenNested` model state is resolved, nested within
+      // a gesture whose completion action requests `awaitNested`.
+    } while(potentialSelector != this.currentSelector);
+
     const selector = this.currentSelector;
 
     touchpoint.setGestureMatchInspector(buildGestureMatchInspector(selector));
+    this.emit('inputstart', touchpoint);
 
-    /* We wait for the source to fully pass through the gesture-model spin-up phase; there's
-     * a chance that the new source will complete an existing gesture instantly without being
-     * locked to it, resulting in activation of a different `stateToken`.
-     *
-     * This, in turn, can affect what the initial 'item' for the new gesture will be.
-     */
-    const modelingSpinupPromise = selector.matchGesture(touchpoint, getGestureModelSet(modelDefs, selector.baseGestureSetId));
-    modelingSpinupPromise.then(async (selectionPromiseHost) => {
-      this.emit('inputstart', touchpoint);
-
-      const selection = await selectionPromiseHost.selectionPromise;
-
-      // Any related 'push' mechanics that may still be lingering are currently handled by GestureSequence
-      // during its 'completion' processing.  (See `GestureSequence.selectionHandler`.)
-      if(selection.result.matched == false) {
+    const selection = await selectionPromise;
+
+    // Any related 'push' mechanics that may still be lingering are currently handled by GestureSequence
+    // during its 'completion' processing.  (See `GestureSequence.selectionHandler`.)
+    if(selection.result.matched == false) {
+      return;
+    }
+
+    // For multitouch gestures, only report the gesture **once**.
+    const sourceIDs = selection.matcher.allSourceIds;
+    for(let sequence of this._activeGestures) {
+      if(!!sequence.allSourceIds.find((id1) => !!sourceIDs.find((id2) => id1 == id2))) {
```

Comment:
O(m*n)? Are these arrays short?

Reply:
A person only has ten fingers, and I'd be quite impressed if someone actually managed to use each of them independently for concurrent gestures.

Secondly, with the way Web currently works, I believe the maximum count for concurrent gestures is 2; a modipress and whatever the current input is on the modipress layer.  Starting a third will immediately resolve the previously-current input (simple tap, flick, multitap) or be blocked (longpress).  There's a _chance_ of it actually being 3 - I'm not 100% sure that a held backspace will be auto-cancelled on new input.  But yeah, that should be it.

This _does_ depend on proper state management of the 'active gestures' array.  If errors happen that prevent removal of finished gesture objects from that array, then over time it may become "less short" than intended.

Reply:
That's what I figured but helpful info on the potential edge cases, thanks!